lib/posix/pthread.c

```diff
@@ -436,24 +433,26 @@ int pthread_join(pthread_t thread, void **status)
 		return ESRCH;
 	}
 
-	pthread_mutex_lock(&pthread->state_lock);
-
-	if (pthread->state == PTHREAD_JOINABLE) {
-		pthread_cond_wait(&pthread->state_cond, &pthread->state_lock);
```

Comment:
this seems like an odd choice -- did the old code really not use `k_thread_join` at all?

Reply:
Yeah... again pthreads were quite different. Almost like "Linux threads" if you remember the old user mode thread implementation that came before nplt